### ISSUE TITLE
SPC / shows "%-4d ..." prompt and fails with ag, grep and ack on MacOS

### ISSUE THREAD

**Issue by stolowski** [syl20bnr/spacemacs#12017]
#### Description :octocat:
SPC / shows "%-4d ..." prompt and fails with ag, grep and ack on MacOS

#### Reproduction guide :beetle:
- Start Emacs
- Open a file
- Hit SPC / (search with ag)

*Observed behaviour:* :eyes: :broken_heart:
The prompt in mini buffer is "%-4d ag from [.....myprojectpath]:. Typing a search string gives "error code 127" in the mini buffer.
It's also broken if I try to use ack (SPC s k f).

*Expected behaviour:* :heart: :smile:
Matching files are listed, no errors displayed, prompt looks sane.

#### System Info :computer:
- OS: darwin
- Emacs: 26.1
- Spacemacs: 0.300.0
- Spacemacs branch: develop (rev. 4b195ddfc)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: ivy
- Layers:
```elisp
(go ivy emacs-lisp git org version-control)
```
- System configuration features: JPEG RSVG IMAGEMAGICK NOTIFY ACL GNUTLS LIBXML2 ZLIB TOOLKIT_SCROLL_BARS NS MODULES THREADS LCMS2

ack version 2.24 (homebrew). ag version 2.2.0 (the_silver_searcher from homebrew). All packages up-to-date as of 2019-02-26.

#### Backtrace :paw_prints:
```
```

**Comment by antonalekseev** [comment]
Have you got a directory that contains `ag` within your `M-: exec-path`?

**Comment by stolowski** [comment]
> Have you got a directory that contains `ag` within your `M-: exec-path`?

Yes, `/usr/local/bin` is there. Also, `SPC /` works with helm ag if I disable ivy in `.spacemacs`.

**Comment by Compro-Prasad** [comment]
Then use `helm-ag` :smile:

**Comment by stolowski** [comment]
> Then use `helm-ag` 😄

Well, sure I can (although I'd prefer ivy), but much nicer would be to fix the problem so the default experience doesn't suffer and other users don't stumble on it - assuming it's a bug and not something specific to my env.

**Comment by Compro-Prasad** [comment]
@stolowski Do `M-x toggle-debug-on-error` first, reproduce the error and report the backtrace.

**Comment by stolowski** [comment]
> @stolowski Do `M-x toggle-debug-on-error` first, reproduce the error and report the backtrace.

Ok, I did that, confirmed with "debug-on-error enabled" in Messages buffer. There is no backtrace, just the broken prompt in the minibuffer and  "error code 127" as soon as I type at least 3 characters.

**Comment by antonalekseev** [comment]
Does `(shell-command-to-string "ag --version")` evaluate and return expected output?

**Comment by stolowski** [comment]
> Does `(shell-command-to-string "ag --version")` evaluate and return expected output?

Bingo! That didn't work; `/usr/local/bin` (where I've ag and ack) is in my PATH (and reflected in `.spacemacs.env`), it is also listed in `exec-path` as reported above, yet it's not seen by `shell-command-to-string`, or by `M-x shell-command`. After some more googling I found out lots of issues around that with emacs on Mac OSX.  TD;LR, I fixed it with:
` (setenv "PATH" (concat (getenv "PATH") ":/usr/local/bin"))` in the user-config section of `.spacemacs`.

Now, the only remaining issue is the "broken" prompt (affecting grep, ag and ack), it still shows:
`%-4d grep from [/Users/pawel/]: `. It seems that the format string for the prompt is incorrect.

**Comment by antonalekseev** [comment]
> Now, the only remaining issue is the "broken" prompt (affecting grep, ag and ack), it still shows:
> `%-4d grep from [/Users/pawel/]: `. It seems that the format string for the prompt is incorrect.

My guess would be that prepending `ivy-count-format` to `ivy-read` prompt

https://github.com/syl20bnr/spacemacs/blob/4b195ddfc9228256361e0b264fe974aa86ed51a8/layers/%2Bcompletion/ivy/funcs.el#L162-L172

is redundant as documenatation for `ivy-read` states:

> PROMPT is a string, normally ending in a colon and a space.
> ‘ivy-count-format’ is prepended to PROMPT during completion.

and count is already prepended to prompt by `ivy-read` itself.

```
1 file changed, 10 insertions(+), 11 deletions(-)
layers/+completion/ivy/funcs.el | 21 ++++++++++-----------

modified   layers/+completion/ivy/funcs.el
@@ -159,17 +159,16 @@ that directory."
             (default-directory
               (or initial-directory (read-directory-name "Start from directory: "))))
       (ivy-read
-       (concat ivy-count-format
-               (format "%s from [%s]: "
-                       tool
-                       (if (< (length default-directory)
-                              spacemacs--counsel-search-max-path-length)
-                           default-directory
-                         (concat
-                          "..." (substring default-directory
-                                           (- (length default-directory)
-                                              spacemacs--counsel-search-max-path-length)
-                                           (length default-directory))))))
+       (format "%s from [%s]: "
+               tool
+               (if (< (length default-directory)
+                      spacemacs--counsel-search-max-path-length)
+                   default-directory
+                 (concat
+                  "..." (substring default-directory
+                                   (- (length default-directory)
+                                      spacemacs--counsel-search-max-path-length)
+                                   (length default-directory)))))
        (spacemacs//make-counsel-search-function tool)
        :initial-input (when initial-input (rxt-quote-pcre initial-input))
        :dynamic-collection t
```

**Comment by yuhan0** [comment]
Is this a Mac OS specific bug? I had the same issue with the "%-4d" prompt some months ago and made a quick workaround (same as @antonalekseev), assuming it was some quirk with my setup.

**Comment by antonalekseev** [comment]
@yuhan0 

> Is this a Mac OS specific bug?

I suppose not. Bug was introduced by https://github.com/abo-abo/swiper/commit/6fe1a938d6b43442f23888219a24a15226a888e4 (#11750 was opened 3 days later), and workaround we both used is appropriate.

**Comment by yuhan0** [comment]
Alright, I went ahead and opened a PR with those changes

**Comment by duianto** [comment]
The PR: Fix ivy prompt #12030
has been cherry-picked into the develop branch.
